funflow-examples/makefile-tool/src/Main.hs

```diff
@@ -105,12 +105,12 @@ buildTarget storeRoot mkfile target@(MakeRule targetNm deps cmd) = let
        depFiles <- flowJoin [Id{ unId = buildTarget storeRoot mkfile r } | r <- depRules] -< (replicate (length depRules) ())
        let fullSrcFiles = Map.fromList $ zip neededSources contentSrcFiles
        -- Compile the target of the current rule.
-       compFile <- (compileFile storeRoot) -< (targetNm, fullSrcFiles, depFiles, cmd)
+       compFile <- caching targetNm (compileFile storeRoot) -< (targetNm, fullSrcFiles, depFiles, cmd)
        returnA -< compFile
 
 
 -- | Compiles a C file in a docker container.
-compileFile :: Path Abs Dir -> Flow (TargetFile, Map.Map SourceFile String, [Content File], Command) (Content File)
+compileFile :: Path Abs Dir -> Flow (TargetFile, Map.Map SourceFile String, [Path Abs File], Command) (Path Abs File)
```

Comment:
Just curious: what was the reason for needing to change the `Flow` input and output types with the introduction of `caching`? As far as I could tell caching should preserve the types of the inputs and outputs. 

https://github.com/tweag/kernmantle/blob/895c46820b7a44f3039de655b6efd73fd6544f39/caching/src/Control/Kernmantle/Caching.hs#L133

Reply:
It was b/c of the `Store` constraint on the output type. I didn't see a hit for an `instance Store` in funflow or in kernmantle for `Content File`, but I did see one for `Path Abs File`.

Reply:
Gotcha that makes sense. Thanks for the additional details.